### ISSUE TITLE
Wine Staging 4.20 not working

### ISSUE THREAD

**Issue by xk0niR** [Tk-Glitch/PKGBUILDS#397]
I can´t use wine staging v4.20 at all while v4.19 works fine for me.
I switched between these two version with
_staging_version="v4.19" /
_staging_version="v4.20"

Since both use the same configuration I don´t think it is an issue with other properties in customization.cfg

If I use v4.20 and just run winecfg (I delete .wine before) I get this output:
wine: created the configuration directory '/home/kon/.wine'
0009:err:environ:run_wineboot failed to start wineboot c00000e5
0009:fixme:actctx:parse_depend_manifests Could not find dependent assembly L"Microsoft.Windows.Common-Controls" (6.0.0.0)
0009:fixme:winediag:start_process Wine TkG 4.20 is a testing version containing experimental patches.
0009:fixme:winediag:start_process Please don't report bugs about it on winehq.org and use https://github.com/Tk-Glitch/PKGBUILDS/issues instead.
0009:err:winecfg:WinMain failed to restart 64-bit L"C:\\windows\\system32\\winecfg.exe", err 1359
0009:err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
0009:err:winediag:nodrv_CreateWindow The explorer process failed to start.

I´ve got no idea whats causing this. If I change to v4.19 winecfg (and games) run fine.
I may could try building commits between these two versions and find the one which causes me problems but I feel like this would take some time.

Thanks a lot!

Edit: It´s also not working in latest master.

**Comment by chorriwuarri** [comment]
for me works but no raw input fix?

**Comment by xk0niR** [comment]
Weird... I´ve compiled again with no different settings and now it works...
I have no idea what the issue was there.
Maybe it´s better to uninstall old wine, reboot and install the new one.

Atleast it seems to be working fine now.I just let this open a bit longer if I get any error soon.

**Comment by Tk-Glitch** [comment]
Weird indeed. Maybe some ccache froggery?

@chorriwuarri The community patch doesn't apply anymore? Or you're talking about it being broken in some ways? The latter is known and the last version reportedly has worse issues so I'm not sure what to do except waiting for it to be improved. You can follow its development and create patchsets against the most recent changes from here if you want: from https://github.com/rbernon/wine/commits/wip/rawinput

**Comment by xDShot** [comment]
Same. Build 4.20, getting this error running on newly created prefix:
```
[xdshot@pbody-pc ~]$ winecfg
000b:fixme:winediag:start_process Wine TkG 4.20 is a testing version containing experimental patches.
000b:fixme:winediag:start_process Please don't report bugs about it on winehq.org and use https://github.com/Tk-Glitch/PKGBUILDS/issues instead.
000d:err:module:__wine_process_init L"C:\\windows\\system32\\winemenubuilder.exe" not found
000b:err:wineboot:process_run_key Error running cmd L"C:\\windows\\system32\\winemenubuilder.exe -r" (126).
0027:fixme:dwmapi:DwmIsCompositionEnabled 00000000068B1518
0032:fixme:iphlpapi:NotifyIpInterfaceChange (family 0, callback 0x2b9d3de, context 0x73fa270, init_notify 0, handle 0x76ff9d0): stub
wine: Unhandled page fault on execute access to 0066D0D8 at address 0066D0D8 (thread 0046), starting debugger...
0046:err:seh:raise_exception Unhandled exception code c0000005 flags 0 addr (nil)
wine: configuration in '/seagate/wineprefixes/steam64' has been updated.
wine: Unhandled page fault on execute access to 0032FAF8 at address 0032FAF8 (thread 0009), starting debugger...
0009:err:seh:raise_exception Unhandled exception code c0000005 flags 0 addr (nil)
[xdshot@pbody-pc ~]$
```

**Comment by telans** [comment]
@xDShot this is a make flag issue, make sure the flags `CROSSCFLAGS` & `CROSSLDFLAGS` the same as your regular flags.

i.e:
```CFLAGS="-march=haswell -O3 -pipe -fno-plt -ftree-vectorize"
CXXFLAGS="${CFLAGS}"
CROSSCFLAGS="${CFLAGS}"
CROSSLDFLAGS="${CFLAGS}"```

**Comment by xDShot** [comment]
@telans shouldn't CROSSLDFLAGS be same as LDFLAGS?..

**Comment by telans** [comment]
I don't believe so, no. Making those four flags the same seems to have fixed that issue for me and another person on the TkG discord

**Comment by xDShot** [comment]
Anywany, I don't think it's somehow related as these variables except CROSSC/LDFLAGS were already same in my makepkg.conf

**Comment by xDShot** [comment]
Yup. 4.19 works. We got regression.

**Comment by telans** [comment]
**_My issue was sort of a duplicate of this, so I'll leave what I posted in that here._**

With recent wine versions (~4.19), compiling with `_LOCAL_OPTIMIZED="false"` and without `CROSSCFLAGS` & `CROSSLDFLAGS` set in makepkg.conf, wine would fail to create a new prefix and run certain 32-bit applications. There would be an error similar to:
```
0009:err:seh:raise_exception Unhandled exception code c0000005 flags 0 addr (nil)
```

xDShot and derandark (discord) have also come across this issue.

I have done a bisect on wine-staging, and it appears the following commit is the first bad one: https://github.com/wine-staging/wine-staging/commit/ef0e88407ea60c9feed923431aee4bcc8ae71f95

I built normal wine against the rebase the commit above was based on. This also had the issue, so whatever caused it is somewhere in here: https://github.com/wine-mirror/wine/compare/070cfc358c8ed67ba08a49ae78c20205ab4e2d00...34ed1e37da1e1c1cfd641f1f16b0e00030cbb357, but I haven't bisected this yet

Whether this is classed as an actual issue or not (maybe the flags should have been set all along?), I think it might be important to add a note about the two make options having to match `CFLAGS` (from what I can tell this is correct, it works for me & others).
